python/hail/methods/statgen.py

```diff
@@ -2929,6 +2929,15 @@ def ld_prune(call_expr, r2=0.2, bp_window_size=1000000, memory_per_core=256, kee
 
     .. include:: ../_templates/req_tvariant.rst
 
+    Examples
+    --------
+    Filter a dataset to those variants returned by ld_prune.
+
+    >>> ds = hl.split_multi_hts(dataset)
```

Comment:
This is going to split the dataset many times. In practice users should either:
- import and split before writing the matrix table (so it's already bi-allelic)
- filter to bi-allelic variants before ld pruning (for array data and most sequencing data, it's still the case that multi-allelics are rare)

So I'd prefer an example like:
```
>>> biallelic_dataset = dataset.filter_rows(hl.len(dataset.alleles) == 2)
>>> pruned_variant_table = hl.ld_prune(biallelic_dataset.GT, r2=0.2, bp_window_size=500000)
>>> filtered_dataset = dataset.filter_rows(hl.is_defined(pruned_variant_table[ds.row_key]))
```

Reply:
why is this going to split many times?

Reply:
this is a very reasonable pipeline

Reply:
And scientifically, I think users would prefer to throw out multi-allelic variants before pruning, since the choice of how to re-code would impact the resulting correlations in perhaps unexpected ways.

Reply:
it'll get split twice inside `ld_prune` and again in `ds.filter_rows`

Reply:
Is it intentional that you have `dataset.filter_rows` instead of `biallelic_dataset.filter_rows`?

Reply:
Both give the same result, but the former won't redo the first filter.